### PR TITLE
v0.1.x: clean up warnings

### PR DESCRIPTION
**Pull request by djc**
## Motivation

Trying to clean up the proliferation of warnings seen with rustc 1.80 (and `--all-features`).

(Best reviewed per commit.)

### CODE REVIEW

File: tracing-core/src/dispatcher.rs [hunk 1]
```diff
@@ -144,12 +144,6 @@ use crate::stdlib::{
     error,
 };
 
-#[cfg(feature = "alloc")]
```
Comment (djc) [inline tracing-core/src/dispatcher.rs:147]: There is no `alloc` features defined in this crate right now.

File: tracing-futures/src/executor/mod.rs [hunk 2]
```diff
@@ -1,12 +1,8 @@
 #[cfg(feature = "futures-01")]
 mod futures_01;
 
-#[cfg(feature = "futures_preview")]
```
Comment (djc) [inline tracing-futures/src/executor/mod.rs:4]: There doesn't appear to be defined feature for `futures_preview`.

File: tracing-subscriber/src/fmt/writer.rs [hunk 3]
```diff
@@ -651,12 +651,6 @@ pub struct Tee<A, B> {
 #[derive(Debug)]
 pub struct MutexGuardWriter<'a, W>(MutexGuard<'a, W>);
 
-/// Implements [`std::io::Write`] for an [`Arc`]<W> where `&W: Write`.
```
Comment (djc) [inline tracing-subscriber/src/fmt/writer.rs:654]: This warned about never being constructed. As far as I can tell it is not actually used for the current impl of `MakeWriter` for `Arc<W>`.

File: tracing-tower/src/lib.rs [hunk 4]
```diff
@@ -70,33 +70,8 @@ where
     }
 }
 
-#[cfg(feature = "tower-util")]
```
Comment (djc) [inline tracing-tower/src/lib.rs:73]: There's no feature defined for `tower-util`.

File: tracing-subscriber/src/fmt/writer.rs (hunk 3 above)
Comment (kaffarell) [reply tracing-subscriber/src/fmt/writer.rs:654]: Hmm I don't get this error. The `ArcWriter` is also public, so it's probably used by someone. I think it's better if don't change this.

**Review by kaffarell — COMMENTED** [review]
LGTM, fixes all the lints/errors I got (except the comment obviously).

File: tracing-subscriber/src/fmt/writer.rs (hunk 3 above)
Comment (djc) [reply tracing-subscriber/src/fmt/writer.rs:654]: It's `pub` within this module, but it's not exported from this module:

```
djc-2021 v0.1.x tracing-subscriber $ rg ArcWriter
src/fmt/writer.rs
658:pub struct ArcWriter<W>(Arc<W>);
1135:// === impl ArcWriter ===
1137:impl<W> io::Write for ArcWriter<W>
```

Comment (hawkw) [reply tracing-subscriber/src/fmt/writer.rs:654]: Ah, whoops; if the `MakeWriter` implementation for `Arc` doesn't actually use this, I think getting rid of it is correct.

**Comment by djc** [comment]
@hawkw so do you want to approve this? (I technically seem to have merge privileges, not sure if you want me to use them.)

File: tracing-subscriber/src/fmt/writer.rs (hunk 3 above)
Comment (kaffarell) [reply tracing-subscriber/src/fmt/writer.rs:654]: > It's `pub` within this module, but it's not exported from this module

Yes it is?
Maybe I'm stupid, but:
https://github.com/tokio-rs/tracing/blob/6d00d7d9f72dc6797138a1062bc33073afbad5a1/tracing-subscriber/src/fmt/writer.rs#L658
and
https://github.com/tokio-rs/tracing/blob/6d00d7d9f72dc6797138a1062bc33073afbad5a1/tracing-subscriber/src/fmt/mod.rs#L201
make it available. Now it probably isn't used anywhere as there's a comment explaining it's an implementation detail.

Comment (djc) [reply tracing-subscriber/src/fmt/writer.rs:654]: Hmm, you're right. So I guess we might want to add `#[doc(hidden)]` on it, or something?

Comment (kaffarell) [reply tracing-subscriber/src/fmt/writer.rs:654]: Sure, and maybe a `#[deprecated(...)]` as well?

Comment (djc) [reply tracing-subscriber/src/fmt/writer.rs:654]: Done.

File: tracing-subscriber/src/lib.rs [hunk 5]
```diff
@@ -240,7 +240,7 @@ feature! {
     #![all(feature = "registry", feature = "std")]
     pub use registry::Registry;
 
-    ///
+    #[allow(missing_docs)]
```
Comment (kaffarell) [inline tracing-subscriber/src/lib.rs:243]: This function is used quite a lot, maybe we should add a small comment – something like "Creates a default Registry"?

Comment (djc) [reply tracing-subscriber/src/lib.rs]: Fair, done.

**Review by kaffarell — APPROVED** [review]
LGTM!

**Review by davidbarsky — APPROVED** [review]
LGTM; would be good to have these on master as well.

**Comment by djc** [comment]
What's the state of master these days? Feels like keeping master for 0.2.x is a little counterproductive if the pace towards actually releasing 0.2.x is low.

**Comment by kaffarell** [comment]
> What's the state of master these days? Feels like keeping master for 0.2.x is a little counterproductive if the pace towards actually releasing 0.2.x is low.

I agree. IMO either we try to release 0.2.x soonish, or we reverse the roles and have 0.1.x -> master and master -> 0.2.x.

**Pull request merged**